### PR TITLE
Adding initial draft of Clair3 WILDS WDL module

### LINKED ISSUE #162: Clair3 Variant Calling Module
During a Data House Call with David Woolston, he mentioned that it would be super beneficial to create a WILDS WDL module for the Clair3 long-read variant caller: https://github.com/HKU-BAL/Clair3

Sounds like he already has the beginnings of some tasks written up, might be an excited opportunity for external contribution and collaboration! 😃 

He also mentioned that ClairS-TO as a another tool by the same developer that might be helpful, not sure if it belongs in the same module or its own separate one: https://github.com/HKU-BAL/ClairS-TO

minimap2
cuteSV
sniffles
svim

### PR DESCRIPTION
**Pull request by tefirman**
## Type of Change

- New module

## Description

Adds a new `ww-clair3` module wrapping [Clair3](https://github.com/HKU-BAL/Clair3), a deep learning-based germline small variant caller from HKU-BAL. Clair3 uses a two-stage pileup and full-alignment approach for high-accuracy SNP and indel calling from ONT, PacBio HiFi, and Illumina sequencing data.

Key features of the module:
- Supports all three sequencing platforms (`ont`, `hifi`, `ilmn`) with platform-specific pre-trained models bundled in the Docker image
- Optional gVCF output for downstream joint genotyping
- GPU acceleration support via `gpu_enabled` parameter
- Region restriction via BED file or contig name

## Related Issue

- Fixes #162 

## Testing

**How did you test these changes?**
Ran linting with `make lint NAME=ww-clair3` (sprocket, miniwdl, WOMtool all pass). Attempting test run currently...

**What workflow engine did you use?**
Sprocket

**Did the tests pass?**
Linting passes on all three engines. Full test run in progress...

## Documentation

- [x] I updated the README (if applicable)
- [x] I added/updated parameter descriptions in the WDL (if applicable)
- [ ] I ran `make docs` to check documentation rendering (if applicable)

## Additional Context

- Uses the newly built `getwilds/clair3:2.0.0` Docker image which bundles all 14 pre-trained models at `/opt/models/` (ONT r9/r10 chemistries, PacBio HiFi Revio/Sequel II, Illumina, and bacteria-finetuned)
- Test workflow uses `platform = "ilmn"` since the `ww-testdata` BAM files are Illumina short-read data
- Dockstore registration in `.dockstore.yml` still needed

### CODE REVIEW

File: modules/ww-clair3/ww-clair3.wdl [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+## for high-accuracy SNP and indel calling from ONT, PacBio, and Illumina reads.
+
+version 1.0
+
+#### TASK DEFINITIONS ####
+
+task run_clair3 {
+  meta {
+    author: "WILDS Team"
+    email: "wilds@fredhutch.org"
```
Comment (emjbishop) [inline modules/ww-clair3/ww-clair3.wdl]: change to your name/email

File: modules/ww-clair3/ww-clair3.wdl [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    memory_gb: "Memory allocated for the task in GB"
+  }
+
+  input {
+    String sample_name
+    File input_bam
+    File input_bam_index
+    File ref_fasta
+    File ref_fasta_index
+    String platform = "ont"
```
Comment (emjbishop) [inline modules/ww-clair3/ww-clair3.wdl]: Is this ONT because we know someone who will use the module for that? Otherwise I'd suggest changing to `ilmn` since that's more common

File: modules/ww-clair3/README.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+Runs Clair3 to call germline variants from aligned reads using deep learning pileup and full-alignment models.
+
+**Inputs:**
+- `sample_name` (String): Name identifier for the sample
+- `input_bam` (File): Aligned reads in BAM format (must be sorted and indexed)
+- `input_bam_index` (File): Index file for the input BAM
+- `ref_fasta` (File): Reference genome FASTA file (must be indexed with .fai)
+- `ref_fasta_index` (File): Index file for the reference FASTA
+- `platform` (String, default="ont"): Sequencing platform: `ont`, `hifi`, or `ilmn`
+- `model_path` (String, default="/opt/models/ont"): Path to Clair3 model directory
```
Comment (emjbishop) [inline modules/ww-clair3/README.md]: Since there are 14 models in here, I would list the paths to all of them in this README so people know what their options are

File: modules/ww-clair3/ww-clair3.wdl [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    File output_vcf_index = "~{sample_name}.clair3.vcf.gz.tbi"
+    Array[File] output_gvcf = glob("*.g.vcf.gz")
+    Array[File] output_gvcf_index = glob("*.g.vcf.gz.tbi")
+  }
+
+  runtime {
+    docker: "getwilds/clair3:2.0.0"
+    cpu: cpu_cores
+    memory: "~{memory_gb} GB"
+    gpus: if gpu_enabled then "1" else "0"
```
Comment (emjbishop) [inline modules/ww-clair3/ww-clair3.wdl:105]:
```suggestion
    gpus: ~{gpu_enabled}
```
Could you do this instead? I don't recall

File: modules/ww-clair3/testrun.wdl [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    model_path = "/opt/models/ilmn",
+    ctg_name = "chr1",
+    pileup_only = true,
+    cpu_cores = 2,
+    memory_gb = 8
+  }
+
+  output {
+    File output_vcf = run_clair3.output_vcf
+    File output_vcf_index = run_clair3.output_vcf_index
```
Comment (emjbishop) [inline modules/ww-clair3/testrun.wdl:52]: I would test out the gVCF functionality also, unless that becomes much more resource intensive

**Review by emjbishop — APPROVED** [review]
This looks great overall, some thoughts below. The key one is just listing all `model_path` options in the README

File: modules/ww-clair3/ww-clair3.wdl (hunk 1 above)
Comment (tefirman) [reply modules/ww-clair3/ww-clair3.wdl]: Great catch, updated in https://github.com/getwilds/wilds-wdl-library/pull/308/commits/cf305866b1402bb908cd196d76c675a52f30c763

File: modules/ww-clair3/ww-clair3.wdl (hunk 2 above)
Comment (tefirman) [reply modules/ww-clair3/ww-clair3.wdl]: Fair point, not sure why I picked `ont` to be honest, adjusted to `ilmn` in https://github.com/getwilds/wilds-wdl-library/pull/308/commits/cf305866b1402bb908cd196d76c675a52f30c763

File: modules/ww-clair3/README.md (hunk 3 above)
Comment (tefirman) [reply modules/ww-clair3/README.md]: Great call, added in https://github.com/getwilds/wilds-wdl-library/pull/308/commits/cf305866b1402bb908cd196d76c675a52f30c763

File: modules/ww-clair3/ww-clair3.wdl (hunk 4 above)
Comment (tefirman) [reply modules/ww-clair3/ww-clair3.wdl:105]: Unfortunately, PROOF does need it to be a string of the number of requested GPU's, something to do with the underlying config. Added a line to the README to explain this in https://github.com/getwilds/wilds-wdl-library/pull/308/commits/cc5143028a3132022fc3e8c0755b2d44fcd09552

File: modules/ww-clair3/testrun.wdl (hunk 5 above)
Comment (tefirman) [reply modules/ww-clair3/testrun.wdl:52]: Super smart, added a gVCF execution to the testrun.wdl in https://github.com/getwilds/wilds-wdl-library/pull/308/commits/cf305866b1402bb908cd196d76c675a52f30c763, just waiting on the CI/CD test run to verify it works.

Comment (tefirman) [reply modules/ww-clair3/testrun.wdl:52]: Made a few fixes in https://github.com/getwilds/wilds-wdl-library/pull/308/commits/843a0ae95856e805211c4bd53830481f97ac3912, looks good now!
